From 0be6542ce700b2810d769ee7e68bc7d6579b7fd6 Mon Sep 17 00:00:00 2001
From: Michael Achenbach <machenbach@chromium.org>
Date: Thu, 14 Jul 2022 09:19:55 +0200
Subject: [PATCH] [test] Skip slow tests

Also bump shards on a slow builder.

No-Try: true
Bug: v8:13074
Change-Id: I07131c53d1d632886039e811de38961dd4eb5752
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3760458
Reviewed-by: Leszek Swirski <leszeks@chromium.org>
Commit-Queue: Leszek Swirski <leszeks@chromium.org>
Auto-Submit: Michael Achenbach <machenbach@chromium.org>
Cr-Commit-Position: refs/heads/main@{#81716}
---
 infra/testing/builders.pyl  | 4 ++--
 test/mjsunit/mjsunit.status | 7 +++++++
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/infra/testing/builders.pyl b/infra/testing/builders.pyl
index 015d5c442fb..0e32c92ce89 100644
--- a/infra/testing/builders.pyl
+++ b/infra/testing/builders.pyl
@@ -1833,7 +1833,7 @@
         'name': 'v8testing',
         'suffix': 'armv8-a',
         'test_args': ['--extra-flags', '--enable-armv8'],
-        'shards': 8
+        'shards': 10
       },
       # Novfp3.
       {
@@ -1854,7 +1854,7 @@
         'suffix': 'novfp3',
         'variant': 'default',
         'test_args': ['--novfp3'],
-        'shards': 8
+        'shards': 3
       },
     ],
   },
diff --git a/test/mjsunit/mjsunit.status b/test/mjsunit/mjsunit.status
index c14bdeee440..1d972ec8301 100644
--- a/test/mjsunit/mjsunit.status
+++ b/test/mjsunit/mjsunit.status
@@ -170,6 +170,7 @@
   'wasm/atomics64-stress': [PASS, SLOW, NO_VARIANTS, ['mode != release or dcheck_always_on', SKIP], ['tsan', SKIP]],
   'wasm/compare-exchange-stress': [PASS, SLOW, NO_VARIANTS],
   'wasm/compare-exchange64-stress': [PASS, SLOW, NO_VARIANTS],
+  'wasm/stringrefs-exec': [PASS, SLOW],
 
   # Very slow on ARM, MIPS, RISCV and LOONG, contains no architecture dependent code.
   'unicode-case-overoptimization0': [PASS, NO_VARIANTS, ['arch in (arm, arm64, mipsel, mips64el, mips64, mips, riscv64, loong64)', SKIP]],
@@ -1274,6 +1275,12 @@
   'regress/wasm/regress-7785': [SKIP],
 }],  # variant == nooptimization
 
+##############################################################################
+['variant in (nooptimization, turboshaft)', {
+  # Slow test:
+  'wasm/stringrefs-exec': [PASS, ['arch in (arm, arm64) and simulator_run and mode == debug', SKIP]],
+}],  # variant in (nooptimization, turboshaft)
+
 ##############################################################################
 ['gcov_coverage', {
   # Tests taking too long.
-- 
2.35.1

